### PR TITLE
Unit test device plugin handler Start()

### PR DESCRIPTION
**Pull request by vikaschoudhary16**
**What this PR does / why we need it**:
This PR adds a UT for device plugin handler's Start() api.
 
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:
This PR is created on top of #55692. Actual LOC is +27.

cc @jeremyeder @sjenning @vishh @jiayingz @RenaudWasTaken @mindprince 
/sig node

### CODE REVIEW

**Review by ScorpioCPH — COMMENTED** [review]
Is this [commit](https://github.com/kubernetes/kubernetes/pull/55698/commits/3eadd1a98a74b8e685d1c2a8d4c8ada314058d55) a duplication of https://github.com/kubernetes/kubernetes/pull/55692

**Comment by dixudx** [comment]
/ok-to-test
@vikaschoudhary16 Please squash your commits. Thanks.

File: pkg/kubelet/cm/deviceplugin/device_plugin_handler_test.go [hunk 1]
```diff
@@ -238,6 +255,27 @@ func (a *activePodsStub) updateActivePods(newPods []*v1.Pod) {
...
+	m, err := NewDevicePluginManagerTestStub()
+	monitorCallback := func(resourceName string, added, updated, deleted []pluginapi.Device) {}
+	testHandler := &HandlerImpl{
+		devicePluginManager:                m,
+		devicePluginManagerMonitorCallback: monitorCallback,
+		allDevices:                         make(map[string]sets.String),
+		allocatedDevices:                   make(map[string]sets.String),
+		podDevices:                         make(podDevices),
+	}
+	as := assert.New(t)
```
Comment (RenaudWasTaken) [inline pkg/kubelet/cm/deviceplugin/device_plugin_handler_test.go]: Can we use the `require package` ?

**Review by RenaudWasTaken — APPROVED** [review]
lgtm

File: pkg/kubelet/cm/deviceplugin/device_plugin_handler_test.go (hunk 1 above)
Comment (vikaschoudhary16) [reply pkg/kubelet/cm/deviceplugin/device_plugin_handler_test.go]: I am still not convinced that why we should be strict in using `require` over `assert`. As you on my another PR:
1.uniformity: If you see both have been used in all the source code.
2. Test would stop immediately with require:  IMO, that depends on individual preference. 
I would rather prefer to keep assert for now.

**Comment by vikaschoudhary16** [comment]
@dixudx done!
Folks, please take a look again!

**Comment by vikaschoudhary16** [comment]
Got superseded by #55730, hence closing!!

**Pull request closed without merging**